pkg/client/strategies_aws.go

```diff
@@ -0,0 +1,266 @@
+/*
+AWS Strategy Setup
+
+A utility to abstract the various strategy map ConfigMaps from the service using CRO (RHMI)
+
+Problem Statement:
+ - We require an AWS strategy map ConfigMaps to be in place to provide configuration used to provision AWS cloud resources
+
+This utility provides the abstraction necessary, provisioning an AWS strategy map
+
+We accept start times for maintenance and backup times as a level of abstraction
+Building the correct maintenance and backup times necessary for AWS
+Maintenance Window format ddd:hh:mm-ddd:hh:mm
+Backup/Snapshot Window hh:mm-hh:mm
+*/
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/rds"
+	croAWS "github.com/integr8ly/cloud-resource-operator/pkg/providers/aws"
+	errorUtil "github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"strings"
+	"time"
+)
+
+const (
+	awsStratName = "cloud-resources-aws-strategies"
```

Comment:
Can we use https://github.com/integr8ly/cloud-resource-operator/blob/master/pkg/providers/aws/config.go#L27 to avoid them becoming out of sync in the future